docs/DeveloperGuide.md

```diff
@@ -250,24 +250,77 @@ The activity diagram shows the general sequence of steps when a user interacts w
+
+5. **Error Handling**
+    - If the required prefix (e.g., `n/`) is missing, the parser throws a `ParseException`.
+    - If an `IOException` or `AccessDeniedException` occurs during data storage, `LogicManager` handles these with an appropriate `CommandException`.
+   
 #### Design considerations
 
+### 1. Predicate Composition
+- **Reason**: Using predicates enables flexible filtering and reduces redundancy by allowing shared filtering logic across commands.
+
+### 2. Case-Insensitive Matching
+- **Reason**: User experience is enhanced by making searches case-insensitive.
+
+
 <br>
 
 ### List person feature
 
 #### Implementation
 
+1. **Input Parsing**  
+   The `list` command input is parsed in `LogicManager` through the `AddressBookParser#parseCommand` method. When the command starts with `list`, the parser invokes `ListCommandParser` to handle the specific arguments.
+
+2. **Entity Type Verification**  
+   `ListCommandParser#parse` trims the input to verify the entity type, checking for `PERSON_ENTITY_STRING` or `APPOINTMENT_ENTITY_STRING`.
+   If the entity type is `PERSON_ENTITY_STRING`, `ListCommandParser` returns an instance of `ListPersonCommand`.
+
+3. **Command Execution**  
+   When executed, `ListPersonCommand#execute` calls `Model#updateFilteredPersonList` with the predicate `PREDICATE_SHOW_ALL_PERSONS` to display all persons.
+
+4. After updating the list, the `ListPersonCommand` returns a success message indicating that all persons have been listed.
+
 #### Design considerations
 
+### 1. Abstracting Common List Logic
+- **Reason**: `ListCommand` is designed as an abstract class to handle shared listing logic for different entities (persons and appointments).
+
 <br>
 
 ### Clear person feature
 
 #### Implementation
 
+1. **Input Parsing**  
+   The `clear` command input is parsed in `LogicManager` by calling `AddressBookParser#parseCommand`. When the command starts with clear, the parser invokes `ClearCommandParser` to process specific arguments.
+
+2. **Entity Type Verification**  
+   `ClearCommandParser#parse` checks if the trimmed input matches `PERSON_ENTITY_STRING` or `APPOINTMENT_ENTITY_STRING`.
+   If the entity type is `PERSON_ENTITY_STRING`, `ClearCommandParser` returns an instance of `ClearPersonCommand`.
+
+3. **Command Execution**  
+   When executed, `ClearPersonCommand#execute` calls `Model#setAddressBook` with a new, empty AddressBook to clear all person entries.
+   After clearing, `ClearPersonCommand` returns a success message indicating the address book has been cleared.
+
 #### Design considerations
 
+### 1. Abstracting Clear Logic
+- **Reason**: The `ClearCommand` abstract class handles shared logic for clearing different entities (person or appointment).
```

Comment:
Should we move this to the start of the section, since the abstraction applies for all the commands?

Reply:
Yes, this abstract logic is common throughout all commands, we should consider making a general design consideration section.